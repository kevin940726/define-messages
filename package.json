{
  "name": "define-messages",
  "version": "0.1.0",
  "description": "ğŸ’¬  define intl messages done right",
  "main": "index.js",
  "repository": "git://github.com/kevin940276/define-messages",
  "author": "Kai Hao",
  "license": "MIT",
  "scripts": {
    "test": "jest"
  },
  "devDependencies": {
    "@babel/core": "~7.0.0-beta.51",
    "@babel/preset-env": "~7.0.0-beta.51",
    "babel-core": "^7.0.0-0",
    "babel-jest": "~23.0.1",
    "jest": "~23.1.0",
    "prettier": "1.13.5"
  }
}
