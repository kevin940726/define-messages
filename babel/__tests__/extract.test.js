import { transform } from "@babel/core";
import { name } from "../../package.json";
import extractPlugin from "../extract";

test("extract to some place", () => {
  const code = `
    import defineMessages from 'define-messages/babel/extract';

    defineMessages({
      id: 'some id',
      defaultMessage: 'some defaultMessage',
    });

    defineMessages({
      title: {
        id: 'title id',
        defaultMessage: 'title defaultMessage',
      },
      description: {
        id: 'description id',
        defaultMessage: 'description defaultMessage',
      },
    });
  `;

  const transformed = transform(code, {
    plugins: [extractPlugin]
  });

  expect(transformed.metadata[name].messages).toMatchSnapshot();
});
