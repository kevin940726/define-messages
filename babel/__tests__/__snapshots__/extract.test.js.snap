// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract to some place 1`] = `
Array [
  Object {
    "defaultMessage": "some defaultMessage",
    "id": "some id",
  },
  Object {
    "defaultMessage": "title defaultMessage",
    "id": "title id",
  },
  Object {
    "defaultMessage": "description defaultMessage",
    "id": "description id",
  },
]
`;
